flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -173,6 +179,10 @@ public ResultPartitionType getPartitionType() {
 		return partitionType;
 	}
 
+	public BufferCompressor getBufferCompressor() {
```

Comment:
Instead of providing getter method, we can provide a utility method to be used by both pipelined and blocking subpartitions. 

```
boolean canBeCompressed(Buffer buffer) {
     return bufferCompressor != null && buffer.isBuffer() && buffer.readableBytes() > 0;
}
```